packages/button/components/ButtonBase.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from "react";
+import { cx } from "emotion";
+import {
+  buttonReset,
+  textWeight,
+  flex,
+  flexItem,
+  display,
+  padding
+} from "../../shared/styles/styleUtils";
+import {
+  button,
+  buttonBase,
+  buttonContent,
+  buttonInverse,
+  fullWidthButton,
+  getInverseMutedButtonStyles,
+  getMutedButtonStyles,
+  processingText
+} from "../style";
+
+export enum ButtonAppearances {
+  Primary = "primary",
+  Secondary = "secondary",
+  Standard = "standard",
+  Danger = "danger",
+  Success = "success"
+}
+
+export interface ButtonProps {
+  /**
+   * if the button triggers new content to appear (e.g.: modals and dropdowns)
+   */
+  ariaHaspopup?: boolean;
+  children?: React.ReactNode | string;
+  /**
+   * whether or not the button is enabled
+   */
+  disabled?: boolean;
+  /**
+   * the icon that appears before the button text
+   */
+  iconStart?: React.ReactElement<HTMLElement>;
+  /**
+   * the icon that appears after the button text
+   */
+  iconEnd?: React.ReactElement<HTMLElement>;
+  /**
+   * whether the action the button was meant to do has completed
+   */
+  isProcessing?: boolean;
+  /**
+   * if the button should fill the width of it's container
+   */
+  isFullWidth?: boolean;
+  /**
+   * if the button is on a dark background
+   */
+  isInverse?: boolean;
```

Comment:
The README didn't mention it because the design team hadn't considered buttons on dark backgrounds until I started working on this ticket. I'll collaborate with Lily and make an update :)

isInverse is just a way to adapt any button styles for a dark background. `StandardButton` and the disabled button style are good examples